src/models/CompanyApplication.js

```diff
@@ -0,0 +1,52 @@
+const mongoose = require("mongoose");
+const { Schema } = mongoose;
+
+const CompanyApplicaionSchema = new Schema({
+    email: {
+        type: String,
+        trim: true,
+        lowercase: true,
+        required: true,
+    },
+    password: { type: String, required: true },
+    companyName: { type: String, required: true },
+    motivation: {
+        type: String,
+        maxlength: 1500,
+        minlength: 10,
+        required: true,
+    },
+    submitedAt: {
+        type: Date,
+        required: true,
+    },
+    approvedAt: {
+        type: Date,
+        validate: [
```

Comment:
I think we should have some type of validation against having approvedAt and rejectedAt set at the same time. They should be mutually exclusive.

Reply:
Agreed, good point